app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -732,18 +739,56 @@ class HomeFragment : Fragment(), CoroutineScope, AccountObserver {
         emitAccountChanges()
     }
 
+    private fun scrollToCollections(addedTabsSize: Int) {
+        launch(Dispatchers.Main) {
```

Comment:
Using Dispatchers.Main directly like this means there is no coroutineScope Job to handle cancellation. The fragment has a coroutineContext using Main already that will cancel and the delay() method will cooperatively cancel when it does so. Is there a reason not to use the current context and to create a new coroutineScope instead?

Reply:
Does this change work @colintheshots ?